src/main/java/cn/nukkit/level/generator/noise/vanilla/d/NoiseGeneratorSimplexD.java

```diff
@@ -0,0 +1,181 @@
+package cn.nukkit.level.generator.noise.vanilla.d;
+
+import cn.nukkit.math.NukkitRandom;
+
+public class NoiseGeneratorSimplexD {
+    public static final double SQRT_3 = Math.sqrt(3.0D);
+    private static final int[][] grad3 = new int[][]{{1, 1, 0}, {-1, 1, 0}, {1, -1, 0}, {-1, -1, 0}, {1, 0, 1}, {-1, 0, 1}, {1, 0, -1}, {-1, 0, -1}, {0, 1, 1}, {0, -1, 1}, {0, 1, -1}, {0, -1, -1}};
+    private static final double F2 = 0.5D * (SQRT_3 - 1.0D);
+    private static final double G2 = (3.0D - SQRT_3) / 6.0D;
+    private final int[] p;
+    public double xo;
+    public double yo;
+    public double zo;
+
+    public NoiseGeneratorSimplexD() {
+        this(new NukkitRandom(System.currentTimeMillis()));
+    }
+
+    public NoiseGeneratorSimplexD(NukkitRandom p_i45471_1_) {
+        this.p = new int[512];
+        this.xo = p_i45471_1_.nextDouble() * 256.0D;
+        this.yo = p_i45471_1_.nextDouble() * 256.0D;
+        this.zo = p_i45471_1_.nextDouble() * 256.0D;
+
+        for (int i = 0; i < 256; this.p[i] = i++) {
+            ;
+        }
+
+        for (int l = 0; l < 256; ++l) {
+            int j = p_i45471_1_.nextBoundedInt(256 - l) + l;
+            int k = this.p[l];
+            this.p[l] = this.p[j];
+            this.p[j] = k;
+            this.p[l + 256] = this.p[l];
+        }
+    }
+
+    private static int fastFloor(double value) {
+        return value > 0.0D ? (int) value : (int) value - 1;
+    }
+
+    private static double dot(int[] p_151604_0_, double p_151604_1_, double p_151604_3_) {
```

Comment:
I think we should probably work on fixing the variable names.

Reply:
already did that on my local branch